code/game/objects/items/weapons/teleprod.dm

```diff
@@ -23,6 +23,9 @@
 
 /obj/item/weapon/melee/baton/cattleprod/attackby(obj/item/I, mob/user, params)//handles sticking a crystal onto a stunprod to make a teleprod
 	..()
+	if(istype(src, /obj/item/weapon/melee/baton/cattleprod/sparkprod))
```

Comment:
Yeah, I wasn't 100% sure how to do this. Is there a way to make this work with only the specified object and not children of it?